### PR TITLE
Resolution for "The path must be absolute" on published projects

### PR DESCRIPTION
**Pull request by Daddoon**
Trying to replace the internal "." folder with the local assembly path in order to have an absolute path for the PhysicalFileStorage class at startup ("The path must be absolute" error at startup for published projects)

May resolve the issue of #376  on dev

_Also apology if (maybe) you see something related to me in commit history on aspnet/Blazor (internally ?), i'm pretty new to Github and Git._

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,4 +5,5 @@ obj/
 launchSettings.json
 artifacts/
 msbuild.binlog
-.vscode/
\ No newline at end of file
+.vscode/
+src/Microsoft.AspNetCore.Blazor.Browser.JS/package-lock.json
```
Comment (SteveSandersonMS) [inline .gitignore]: Could you revert the changes to this file? We don't want to exclude `package-lock.json` from source control.

File: src/Microsoft.AspNetCore.Blazor.Server/BlazorConfig.cs [hunk 2]
```diff
@@ -29,6 +29,9 @@ private BlazorConfig(string assemblyPath)
             var configLines = File.ReadLines(configFilePath).ToList();
             SourceMSBuildPath = configLines[0];
 
+            if (string.IsNullOrEmpty(SourceMSBuildPath) || SourceMSBuildPath == ".")
```
Comment (SteveSandersonMS) [inline src/Microsoft.AspNetCore.Blazor.Server/BlazorConfig.cs]: Is there any scenario where `SourceMSBuildPath` is null or empty? I'm not aware that there is. If I'm correct, could you revert that part of the `if` clause?

File: src/Microsoft.AspNetCore.Blazor.Server/BlazorConfig.cs [hunk 3]
```diff
@@ -29,6 +29,9 @@ private BlazorConfig(string assemblyPath)
...
+                SourceMSBuildPath = assemblyPath;
```
Comment (SteveSandersonMS) [inline src/Microsoft.AspNetCore.Blazor.Server/BlazorConfig.cs]: Could you put braces on separate lines around this one? The code style here is always to write:

```
if (condition)
{
    action();
}
```

... and not just

```
if (condition)
    action();
```

File: src/Microsoft.AspNetCore.Blazor.Server/BlazorConfig.cs (hunk 2 above)
Comment (Daddoon) [reply src/Microsoft.AspNetCore.Blazor.Server/BlazorConfig.cs]: Actualy no, i have only seen "." instead, maybe i have overdone this. We can assume that "." case is the only alternate scenario possible actually.

I will modify this part, thanks !

File: src/Microsoft.AspNetCore.Blazor.Server/BlazorConfig.cs (hunk 3 above)
Comment (Daddoon) [reply src/Microsoft.AspNetCore.Blazor.Server/BlazorConfig.cs]: Thanks for the information, i will do it like this in futur contribution on the project.

**Comment by SteveSandersonMS** [comment]
Thanks @Daddoon! This is now merged into `dev`.
